lang/src/org/partiql/lang/eval/ExprValueExtensions.kt

```diff
@@ -227,7 +227,8 @@ private val genericTimeRegex = Regex("\\d\\d:\\d\\d:\\d\\d(\\.\\d*)?([+|-]\\d\\d
 fun ExprValue.cast(
     targetDataType: DataType,
     valueFactory: ExprValueFactory,
-    locationMeta: SourceLocationMeta?
+    locationMeta: SourceLocationMeta?,
+    env: Environment
```

Comment:
I think you should be passing `EvaluationSession` here, not `Environment`.  `Environment` has a lot of details related to the current state of the interpreter that aren't relevant to casting.  Generally, we want to avoid sharing those details as much as reasonably possible.

Reply:
Made the change in the new commit.